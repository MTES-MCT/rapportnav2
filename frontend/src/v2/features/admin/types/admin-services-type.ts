export interface AdminService {
  id: number
  name: String
  createdAt?: Date
  updatedAt?: Date
  controlUnitId?: number
}

export type AdminAction = 'DELETE' | 'CREATE' | 'UPDATE'
export type AdminCell = { key: string; label: string; width: number }
