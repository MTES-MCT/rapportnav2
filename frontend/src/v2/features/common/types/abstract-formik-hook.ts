import { FormikErrors } from 'formik'
import { ObjectSchema } from 'yup'

export interface AbstractFormikHook<T, M> {
  initValue?: M
  errors?: FormikErrors<M>
  beforeInitValue: (value?: T) => M | undefined
  beforeSubmit: (value?: M, errors?: FormikErrors<M>) => T | undefined
  handleSubmit: (
    value?: M,
    errors?: FormikErrors<M>,
    onSubmit?: (valueToSubmit?: T) => Promise<unknown>
  ) => Promise<void>
}

export interface AbstractFormikSubFormHook<M> {
  initValue?: M
  errors?: FormikErrors<M>
  validationSchema?: ObjectSchema<any>
  handleSubmit: (value?: M, errors?: FormikErrors<M>) => Promise<void>
}
