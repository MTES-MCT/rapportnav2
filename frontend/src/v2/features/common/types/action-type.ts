export enum ActionType {
  CONTROL = 'CONTROL',
  NOTE = 'NOTE',
  SURVEILLANCE = 'SURVEILLANCE',
  STATUS = 'STATUS',
  CONTACT = 'CONTACT',
  RESCUE = 'RESCUE',
  OTHER = 'OTHER',
  NAUTICAL_EVENT = 'NAUTICAL_EVENT',
  VIGIMER = 'VIGIMER',
  ANTI_POLLUTION = 'ANTI_POLLUTION',
  BAAEM_PERMANENCE = 'BAAEM_PERMANENCE',
  PUBLIC_ORDER = 'PUBLIC_ORDER',
  REPRESENTATION = 'REPRESENTATION',
  ILLEGAL_IMMIGRATION = 'ILLEGAL_IMMIGRATION',
  INQUIRY = 'INQUIRY',
  COMMUNICATION = 'COMMUNICATION',
  TRAINING = 'TRAINING',
  UNIT_MANAGEMENT_PLANNING = 'UNIT_MANAGEMENT_PLANNING',
  UNIT_MANAGEMENT_TRAINING = 'UNIT_MANAGEMENT_TRAINING',
  UNIT_MANAGEMENT_OTHER = 'UNIT_MANAGEMENT_OTHER',
  CONTROL_SECTOR = 'CONTROL_SECTOR',
  CONTROL_NAUTICAL_LEISURE = 'CONTROL_NAUTICAL_LEISURE',
  CONTROL_SLEEPING_FISHING_GEAR = 'CONTROL_SLEEPING_FISHING_GEAR',
  OTHER_CONTROL = 'OTHER_CONTROL',
  RESOURCES_MAINTENANCE = 'RESOURCES_MAINTENANCE',
  MEETING = 'MEETING',
  PV_DRAFTING = 'PV_DRAFTING',
  HEARING_CONDUCT = 'HEARING_CONDUCT',
  LAND_SURVEILLANCE = 'LAND_SURVEILLANCE',
  FISHING_SURVEILLANCE = 'FISHING_SURVEILLANCE',
  MARITIME_SURVEILLANCE = 'MARITIME_SURVEILLANCE'
}

export enum ActionGroupType {
  OTHER_GROUP = 'OTHER_GROUP',
  CONTROL_GROUP = 'CONTROL_GROUP',
  SURVEILLANCE_GROUP = 'SURVEILLANCE_GROUP',
  ADMINISTRATIVE_GROUP = 'ADMINISTRATIVE_GROUP',
  UNIT_MANAGEMENT_GROUP = 'UNIT_MANAGEMENT_GROUP'
}
