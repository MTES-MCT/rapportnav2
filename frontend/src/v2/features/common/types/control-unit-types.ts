export type ControlUnitResource = {
  id: string
  controlUnit: ControlUnit
  controlUnitId: number
  isArchived: boolean
  name: string
  note?: string
  photo?: string
  station: Station
  stationId: number
  type: ControlUnitResourceType
}

export type Station = {
  id: number
  latitude: number
  longitude: number
  name: string
}

export type ControlUnit = {
  id: number
  areaNote?: string
  administrationId: number
  departmentAreaInseeCode?: string
  isArchived: boolean
  name: string
  termsNote?: string
}

export type Administration = {
  id: number
  name: string
  controlUnitIds: number[]
  isArchived: boolean
  controlUnits: ControlUnit[]
}

enum ControlUnitResourceType {
  AIRPLANE = "AIRPLANE",
  BARGE = "BARGE",
  CAR = "CAR",
  DRONE = "DRONE",
  EQUESTRIAN = "EQUESTRIAN",
  FAST_BOAT = "FAST_BOAT",
  FRIGATE = "FRIGATE",
  HELICOPTER = "HELICOPTER",
  HYDROGRAPHIC_SHIP = "HYDROGRAPHIC_SHIP",
  KAYAK = "KAYAK",
  LIGHT_FAST_BOAT = "LIGHT_FAST_BOAT",
  MINE_DIVER = "MINE_DIVER",
  MOTORCYCLE = "MOTORCYCLE",
  NET_LIFTER = "NET_LIFTER",
  NO_RESOURCE = "NO_RESOURCE",
  OTHER = "OTHER",
  PATROL_BOAT = "PATROL_BOAT",
  PEDESTRIAN = "PEDESTRIAN",
  PIROGUE = "PIROGUE",
  RIGID_HULL = "RIGID_HULL",
  SEA_SCOOTER = "SEA_SCOOTER",
  SEMI_RIGID = "SEMI_RIGID",
  SUPPORT_SHIP = "SUPPORT_SHIP",
  TRAINING_SHIP = "TRAINING_SHIP",
  TUGBOAT = "TUGBOAT"
}
