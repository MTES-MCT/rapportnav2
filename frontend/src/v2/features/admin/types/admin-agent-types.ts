import { AdminService } from './admin-services-type'

export type Agent = {
  id?: string
  firstName?: string
  lastName?: string
  services?: string[]
  deletedAt?: Date
  createdAt?: Date
  updatedAt?: Date
}

export type AgentRole = {
  id: string
  title: string
}

export type AdminAgentService = {
  id: number
  agent: Agent
  role: AgentRole
  createdAt?: Date
  updatedAt?: Date
  disabledAt?: Date
}

export type AdminServiceWithAgent = {
  service: AdminService
  agentServices: AdminAgentService[]
}

export type AdminAgentServiceInput = {
  id?: number
  agentId: number
  serviceId: number
  roleId?: number
}
