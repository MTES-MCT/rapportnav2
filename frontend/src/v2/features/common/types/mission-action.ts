import { ActionStatusReason, ActionStatusType } from '@common/types/action-types'
import { ControlMethod, ControlType } from '@common/types/control-types'
import { ActionTargetTypeEnum, VehicleTypeEnum, VesselSizeEnum, VesselTypeEnum } from '@common/types/env-mission-types'
import {
  ControlCheck,
  FleetSegment,
  GearControl,
  GearInfraction,
  LogbookInfraction,
  MissionActionType,
  OtherInfraction,
  SpeciesControl,
  SpeciesInfraction
} from '@common/types/fish-mission-types'
import { InfractionByTarget } from '@common/types/infraction-types'
import { CompletenessForStats } from '@common/types/mission-types'
import { ActionType } from './action-type'
import { CrossControl } from './crossed-control-type.ts'
import { MissionActionData } from './mission-action-data'
import { MissionSource } from './mission-types'
import { NetworkSyncStatus } from './network-types.ts'

export interface MissionAction {
  id?: string
  missionId: number
  actionType: ActionType
  source: MissionSource
  status?: ActionStatusType
  summaryTags?: string[]
  isCompleteForStats?: boolean
  controlsToComplete: ControlType[]
  completenessForStats: CompletenessForStats
  sourcesOfMissingDataForStats: MissionSource[]
  networkSyncStatus?: NetworkSyncStatus
  data: MissionActionData
}

export interface MissionNavActionData extends MissionActionData {
  detectedPollution?: boolean
  pollutionObservedByAuthorizedAgent?: boolean
  diversionCarriedOut?: boolean
  isSimpleBrewingOperationDone?: boolean
  isAntiPolDeviceDeployed?: boolean
  controlMethod?: ControlMethod
  vesselIdentifier?: string
  vesselType?: VesselTypeEnum
  vesselSize?: VesselSizeEnum
  identityControlledPerson?: string
  nbOfnumbererceptedVessels?: number
  nbOfnumbererceptedMigrants?: number
  nbOfSuspectedSmugglers?: number
  isVesselRescue?: boolean
  isPersonRescue?: boolean
  isVesselNoticed?: boolean
  isVesselTowed?: boolean
  isInSRRorFollowedByCROSSMRCC?: boolean
  numberPersonsRescued?: number
  numberOfDeaths?: number
  operationFollowsDEFREP?: boolean
  locationDescription?: string
  isMigrationRescue?: boolean
  nbOfVesselsTrackedWithoutnumberervention?: number
  nbAssistedVesselsReturningToShore?: number
  reason?: ActionStatusReason
  crossControl?: CrossControl
}

export interface MissionNavAction extends MissionAction {
  data: MissionNavActionData
}

export interface MissionEnvActionData extends MissionActionData {
  completedBy: string
  geom: string
  facade: string
  department: string
  isAdministrativeControl: Boolean
  isComplianceWithWaterRegulationsControl: Boolean
  isSafetyEquipmentAndStandardsComplianceControl: Boolean
  isSeafarersControl: Boolean
  openBy: string
  observations: string
  observationsByUnit: string
  actionNumberOfControls: number
  actionTargetType: ActionTargetTypeEnum
  vehicleType: VehicleTypeEnum
  infractions: InfractionByTarget[]
  coverMissionZone: Boolean
  formattedControlPlans: any
  controlsToComplete: ControlType[]
  availableControlTypesForInfraction: ControlType[]
}

export interface MissionEnvAction extends MissionAction {
  data: MissionEnvActionData
}

export interface MissionFishActionData extends MissionActionData {
  vesselId?: number
  vesselName?: string
  internalReferenceNumber?: string
  externalReferenceNumber?: string
  districtCode?: string
  faoAreas: string[]
  fishActionType: MissionActionType
  emitsVms?: ControlCheck
  emitsAis?: ControlCheck
  logbookMatchesActivity?: ControlCheck
  licencesMatchActivity?: ControlCheck
  speciesWeightControlled?: boolean
  speciesSizeControlled?: boolean
  separateStowageOfPreservedSpecies?: ControlCheck
  logbookInfractions?: LogbookInfraction[]
  licencesAndLogbookObservations?: string
  gearInfractions?: GearInfraction[]
  speciesInfractions?: SpeciesInfraction[]
  speciesObservations?: string
  seizureAndDiversion?: boolean
  otherInfractions?: OtherInfraction[]
  numberOfVesselsFlownOver?: number
  unitWithoutOmegaGauge?: boolean
  controlQualityComments?: string
  feedbackSheetRequired?: boolean
  userTrigram: string
  segments?: FleetSegment[]
  facade?: string
  longitude?: number
  latitude?: number
  portLocode?: string
  portName?: string
  vesselTargeted?: ControlCheck
  seizureAndDiversionComments?: string
  otherComments?: string
  gearOnboard?: GearControl[]
  speciesOnboard?: SpeciesControl[]
  isFromPoseidon?: boolean
  isDeleted?: boolean
  hasSomeGearsSeized?: boolean
  hasSomeSpeciesSeized?: boolean
  completedBy?: string
  completion?: string
  isAdministrativeControl?: boolean
  isComplianceWithWaterRegulationsControl?: boolean
  isSafetyEquipmentAndStandardsComplianceControl?: boolean
  isSeafarersControl?: boolean
  observationsByUnit?: string
  speciesQuantitySeized?: number
}

export interface MissionFishAction extends MissionAction {
  data: MissionFishActionData
}
