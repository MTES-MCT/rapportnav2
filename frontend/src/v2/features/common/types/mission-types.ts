import { ControlResource, ControlUnit } from '@common/types/control-unit-types.ts'
import { MissionCrew } from '@common/types/crew-types'
import { MissionTypeEnum, SeaFrontEnum } from '@common/types/env-mission-types'
import { Service } from '@common/types/service-types'
import { MissionAction } from './mission-action'
import { ControlUnitResource } from './control-unit-types.ts'

export enum MissionType {
  AIR = 'AIR',
  LAND = 'LAND',
  SEA = 'SEA'
}

export type MissionGeneralInfoPam = {
  id?: number
  distanceInNauticalMiles?: number
  consumedGOInLiters?: number
  consumedFuelInLiters?: number
  serviceId?: number
  nbrOfRecognizedVessel?: number
}

export type MissionULAMGeneralInfoInitial = {
  id?: number
  startDateTimeUtc: string
  endDateTimeUtc: string
  missionTypes: MissionTypeEnum[]
  missionReportType?: MissionReportTypeEnum
  reinforcementType?: MissionReinforcementTypeEnum
  nbHourAtSea?: number
}

export type MissionGeneralInfoExtended = {
  resources?: ControlResource[]
  crew?: MissionCrew[]
  isMissionArmed?: boolean
  isWithInterMinisterialService?: boolean
  observations?: string
  isAllAgentsParticipating?: boolean
  interMinisterialServices?: InterMinisterialService[]
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export enum MissionReportTypeEnum {
  FIELD_REPORT = 'FIELD_REPORT',
  OFFICE_REPORT = 'OFFICE_REPORT',
  EXTERNAL_REINFORCEMENT_TIME_REPORT = 'EXTERNAL_REINFORCEMENT_TIME_REPORT'
}

export enum JdpTypeEnum {
  DOCKED = 'DOCKED',
  ONBOARD = 'ONBOARD',
}

export enum MissionReinforcementTypeEnum {
  PATROL = 'PATROL',
  OTHER_ULAM = 'OTHER_ULAM',
  SEA_TRAINER = 'SEA_TRAINER',
  OTHER = 'OTHER',
  DIRM = 'DIRM'
}

export enum MissionSourceEnum {
  MONITORENV = 'MONITORENV',
  MONITORFISH = 'MONITORFISH',
  POSEIDON_CACEM = 'POSEIDON_CACEM',
  POSEIDON_CNSP = 'POSEIDON_CNSP',
  RAPPORTNAV = 'RAPPORTNAV'
}

export enum CompletenessForStatsStatusEnum {
  COMPLETE = 'COMPLETE',
  INCOMPLETE = 'INCOMPLETE'
}

export enum MissionStatusEnum {
  UPCOMING = 'UPCOMING',
  IN_PROGRESS = 'IN_PROGRESS',
  ENDED = 'ENDED',
  UNAVAILABLE = 'UNAVAILABLE'
}

export type Mission = {
  id?: number
  missionTypes: MissionType[]
  controlUnits: ControlUnit[]
  openBy?: string
  completedBy?: string
  observationsCacem?: string
  observationsCnsp?: string
  facades?: string
  geom?: string
  startDateTimeUtc: string
  endDateTimeUtc?: string
  missionSource: MissionSourceEnum
  hasMissionOrder: boolean
  isUnderJdp: boolean
  isGeometryComputedFromControls: boolean
}

export type CompletenessForStats = {
  sources?: MissionSourceEnum[]
  status?: CompletenessForStatsStatusEnum
}

export type MissionEnvData = {
  controlUnits: Omit<ControlUnit, 'id'>[]
  endDateTimeUtc?: string
  facade: SeaFrontEnum
  geom?: Record<string, any>[]
  hasMissionOrder?: boolean
  isUnderJdp?: boolean
  missionSource: MissionSourceEnum
  missionTypes: MissionTypeEnum[]
  observationsCacem?: string
  observationsCnsp?: string
  observationsByUnit?: string
  openBy: string
  startDateTimeUtc: string
}

export type MissionGeneralInfo2 = MissionGeneralInfoPam & {
  missionReportType?: MissionReportTypeEnum
  missionTypes: MissionTypeEnum[]
  reinforcementType?: MissionReinforcementTypeEnum
  nbHourAtSea?: number
  crew?: MissionCrew[]
  services?: Service[]
  isMissionArmed?: boolean
  isWithInterMinisterialService?: boolean
  isAllAgentsParticipating?: boolean
  missionId?: number
  startDateTimeUtc: string
  endDateTimeUtc: string
  observations?: string
  resources?: ControlResource[]
  interMinisterialServices?: InterMinisterialService[]
  missionIdString?: string
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export type Mission2 = {
  id: number
  isCompleteForStats: boolean
  status: MissionStatusEnum
  envData: MissionEnvData
  generalInfos: MissionGeneralInfo2
  completenessForStats?: CompletenessForStats
  actions: MissionAction[]
  missionIdString?: string
}

export type MissionListItem = {
  id: number
  openBy?: string
  exportLabel?: string
  status?: MissionStatusEnum
  missionNameUlam?: string
  startDateTimeUtc: string
  startDateTimeUtcText?: string
  crew?: MissionCrew[]
  crewNumber?: string
  endDateTimeUtc?: string
  endDateTimeUtcText?: string
  missionNamePam?: string
  observationsByUnit?: string
  missionSource?: MissionSourceEnum
  completenessForStats?: CompletenessForStats
  controlUnits?: ControlUnit[]
  resources?: ControlUnitResource[],
  missionReportType?: MissionReportTypeEnum
  missionIdString?: string
  isUnderJdp?: boolean
  jdpType?: JdpTypeEnum
}

export type InterMinisterialService = {
  administrationId: number
  controlUnitId: number
}
