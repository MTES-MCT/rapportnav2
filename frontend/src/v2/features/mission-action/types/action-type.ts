import { MissionEnvActionData, MissionFishActionData, MissionNavActionData } from '../../common/types/mission-action'
import { MissionActionData } from '../../common/types/mission-action-data'
import { RescueType } from '../../common/types/rescue-type'

export type ActionGenericDateObservation = {
  id: string
  startDateTimeUtc: string
  endDateTimeUtc: string
  observations: string
}

export type ActionGenericDateObservationInput = {
  dates: Date[]
} & MissionActionData

export type ActionAntiPollutionInput = {
  dates: Date[]
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionIllegalImmigrationInput = {
  dates: Date[]
  isMissionFinished?: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionFreeNoteInput = {
  date: Date
} & MissionNavActionData

export type ActionRescueInput = {
  dates: Date[]
  rescueType: RescueType
  isMissionFinished?: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionStatusInput = {
  date: Date
} & MissionNavActionData

export type ActionSurveillanceInput = {
  dates: Date[]
} & MissionEnvActionData

export type ActionNavControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: (number | undefined)[]
} & MissionNavActionData

export type ActionFishControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: (number | undefined)[]
} & MissionFishActionData

export type ActionEnvControlInput = {
  dates: Date[]
  isMissionFinished: boolean
  geoCoords: [number?, number?]
} & MissionEnvActionData
