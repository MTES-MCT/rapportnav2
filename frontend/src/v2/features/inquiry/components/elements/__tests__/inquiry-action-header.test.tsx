import { ActionStatusType } from '@common/types/action-types'
import { MissionSourceEnum } from '@common/types/env-mission-types'
import { CompletenessForStatsStatusEnum } from '@common/types/mission-types'
import { useStore } from '@tanstack/react-store'
import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import { ActionType } from '../../../../common/types/action-type'
import { MissionAction } from '../../../../common/types/mission-action'
import InquiryActionHeader from '../inquiry-action-header'

const action: MissionAction = {
  source: MissionSourceEnum.RAPPORTNAV,
  id: 'nav-id',
  missionId: '2',
  actionType: ActionType.INQUIRY,
  isCompleteForStats: false,
  status: ActionStatusType.ANCHORED,
  summaryTags: ['tag2'],
  controlsToComplete: [],
  completenessForStats: { status: CompletenessForStatsStatusEnum.INCOMPLETE },
  data: {
    startDateTimeUtc: '2025-01-22T10:00:00Z',
    endDateTimeUtc: '2025-01-22T11:00:00Z',
    observations: 'nav obs',
    controlMethod: 'method',
    vesselIdentifier: 'id',
    vesselType: 'type',
    vesselSize: 'size',
    isVesselRescue: true,
    isPersonRescue: false,
    reason: 'reason',
    nbrOfHours: 2
  }
}

// Mock dependencies
vi.mock('@tanstack/react-store', () => ({
  useStore: vi.fn()
}))

vi.mock('../../hooks/use-inquiry-timeline-registry', () => ({
  getTimeline: () => ({
    icon: 'ðŸ“‹',
    title: 'Action'
  })
}))

vi.mock('../../hooks/use-inquiry-timeline-registry')
vi.mock('../../../common/components/layout/action-header-wrapper', () => ({
  default: ({ title, icon }: any) => (
    <div data-testid="header-wrapper">
      {title} {icon}
    </div>
  )
}))

vi.mock('../../../../common/components/elements/action-header-action', () => ({
  default: (props: any) => <div data-testid="header-wrapper">Action Header ....</div>
}))

describe('InquiryActionHeader', () => {
  beforeEach(() => {
    vi.mocked(useStore).mockReturnValue(1)
  })

  it('renders with required props', () => {
    const wrapper = render(<InquiryActionHeader inquiryId="123" action={action} />)
    expect(wrapper).toMatchSnapshot()
  })
})
