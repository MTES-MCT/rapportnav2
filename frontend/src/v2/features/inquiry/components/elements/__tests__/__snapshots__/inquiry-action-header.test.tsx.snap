// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InquiryActionHeader > renders with required props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-stack"
        style="align-items: center; flex-direction: column; gap: 0.5rem;"
      >
        <div
          class="rs-stack-item"
          style="width: 100%;"
        >
          <div
            class="rs-stack"
            style="align-items: initial; flex-direction: row; gap: 0.5rem; width: 100%;"
          >
            <div
              class="rs-stack-item"
              style="align-self: baseline;"
            >
              <span
                class="sc-hjsuWn CTYBx Element-IconBox"
              >
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <g
                    transform="translate(80 205)"
                  >
                    <rect
                      fill="currentColor"
                      height="12"
                      transform="translate(-64 -188) rotate(90)"
                      width="2"
                    />
                    <path
                      d="M-67.694-199l1.333,8h-7.277l1.336-8Zm0-2h-4.6a2,2,0,0,0-1.977,1.675L-76-189h12l-1.721-10.325A2,2,0,0,0-67.7-201Z"
                      fill="currentColor"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-70.75 -204.25)"
                      width="1.5"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-64.061 -203) rotate(45)"
                      width="1.5"
                    />
                    <rect
                      fill="currentColor"
                      height="2.25"
                      transform="translate(-77 -201.939) rotate(-45)"
                      width="1.5"
                    />
                  </g>
                  <rect
                    fill="none"
                    height="20"
                    width="20"
                  />
                </svg>
              </span>
            </div>
            <div
              class="rs-stack-item"
              style="flex-grow: 2;"
            >
              <div
                class="rs-stack"
                style="align-items: center; flex-direction: row; flex-wrap: wrap; gap: 0.5rem;"
              >
                <div
                  class="rs-stack-item"
                >
                  <p
                    class="sc-iRcyzz sc-enPhjR jUHdyB kakyJF"
                    font-size="16"
                  >
                    Session de travail 1
                  </p>
                </div>
                <div
                  class="rs-stack-item"
                >
                  <p
                    class="sc-iRcyzz sc-enPhjR jUHdyB gRgWTP"
                    font-size="16"
                    weight="normal"
                  >
                    (
                    22 janv. à 11:00
                    )
                  </p>
                </div>
              </div>
            </div>
            <div
              class="rs-stack-item"
            >
              <div
                data-testid="header-wrapper"
              >
                Action Header ....
              </div>
            </div>
          </div>
        </div>
        <div
          class="rs-stack-item"
          style="width: 100%;"
        >
          <div
            class="rs-stack"
            style="align-items: center; gap: 0.5rem;"
          >
            <div
              class="rs-stack-item"
              style="align-self: center; color: rgb(59, 69, 89);"
            >
              <span
                class="sc-hjsuWn eVTFbL Element-IconBox"
              >
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <g
                    transform="translate(200 -82)"
                  >
                    <path
                      d="M-190,83a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-190,83Z"
                      fill="currentColor"
                    />
                    <rect
                      fill="#fff"
                      height="5"
                      transform="translate(-189 93) rotate(180)"
                      width="2"
                    />
                    <rect
                      fill="#fff"
                      height="2"
                      transform="translate(-189 96) rotate(180)"
                      width="2"
                    />
                  </g>
                  <rect
                    fill="none"
                    height="20"
                    width="20"
                  />
                </svg>
              </span>
            </div>
            <div
              class="rs-stack-item"
              style="align-self: baseline;"
            >
              <p
                class="sc-iRcyzz sc-fFmWCW jUHdyB hAFmCO"
                color="#3B4559"
                font-size="13"
              >
                Des champs indispensables sont à remplir par .
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-stack"
      style="align-items: center; flex-direction: column; gap: 0.5rem;"
    >
      <div
        class="rs-stack-item"
        style="width: 100%;"
      >
        <div
          class="rs-stack"
          style="align-items: initial; flex-direction: row; gap: 0.5rem; width: 100%;"
        >
          <div
            class="rs-stack-item"
            style="align-self: baseline;"
          >
            <span
              class="sc-hjsuWn CTYBx Element-IconBox"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <g
                  transform="translate(80 205)"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    transform="translate(-64 -188) rotate(90)"
                    width="2"
                  />
                  <path
                    d="M-67.694-199l1.333,8h-7.277l1.336-8Zm0-2h-4.6a2,2,0,0,0-1.977,1.675L-76-189h12l-1.721-10.325A2,2,0,0,0-67.7-201Z"
                    fill="currentColor"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-70.75 -204.25)"
                    width="1.5"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-64.061 -203) rotate(45)"
                    width="1.5"
                  />
                  <rect
                    fill="currentColor"
                    height="2.25"
                    transform="translate(-77 -201.939) rotate(-45)"
                    width="1.5"
                  />
                </g>
                <rect
                  fill="none"
                  height="20"
                  width="20"
                />
              </svg>
            </span>
          </div>
          <div
            class="rs-stack-item"
            style="flex-grow: 2;"
          >
            <div
              class="rs-stack"
              style="align-items: center; flex-direction: row; flex-wrap: wrap; gap: 0.5rem;"
            >
              <div
                class="rs-stack-item"
              >
                <p
                  class="sc-iRcyzz sc-enPhjR jUHdyB kakyJF"
                  font-size="16"
                >
                  Session de travail 1
                </p>
              </div>
              <div
                class="rs-stack-item"
              >
                <p
                  class="sc-iRcyzz sc-enPhjR jUHdyB gRgWTP"
                  font-size="16"
                  weight="normal"
                >
                  (
                  22 janv. à 11:00
                  )
                </p>
              </div>
            </div>
          </div>
          <div
            class="rs-stack-item"
          >
            <div
              data-testid="header-wrapper"
            >
              Action Header ....
            </div>
          </div>
        </div>
      </div>
      <div
        class="rs-stack-item"
        style="width: 100%;"
      >
        <div
          class="rs-stack"
          style="align-items: center; gap: 0.5rem;"
        >
          <div
            class="rs-stack-item"
            style="align-self: center; color: rgb(59, 69, 89);"
          >
            <span
              class="sc-hjsuWn eVTFbL Element-IconBox"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <g
                  transform="translate(200 -82)"
                >
                  <path
                    d="M-190,83a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-190,83Z"
                    fill="currentColor"
                  />
                  <rect
                    fill="#fff"
                    height="5"
                    transform="translate(-189 93) rotate(180)"
                    width="2"
                  />
                  <rect
                    fill="#fff"
                    height="2"
                    transform="translate(-189 96) rotate(180)"
                    width="2"
                  />
                </g>
                <rect
                  fill="none"
                  height="20"
                  width="20"
                />
              </svg>
            </span>
          </div>
          <div
            class="rs-stack-item"
            style="align-self: baseline;"
          >
            <p
              class="sc-iRcyzz sc-fFmWCW jUHdyB hAFmCO"
              color="#3B4559"
              font-size="13"
            >
              Des champs indispensables sont à remplir par .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
